### PR TITLE
dsc: make driver compile

### PR DESCRIPTION
**Pull request by kcxt**
Fixes a few compile errors. Doesn't give me a working panel yet

### CODE REVIEW

**Comment by MarijnS95** [comment]
Almost missed this PR, turns out forking a repo (through `gh repo fork`) doesn't make myself watch my own repo anymore (??).

@calebccff Do you mind PR'ing this entire DSC feature branch upstream instead? I don't use the panel generator myself and won't be able to vet/test these changes. Do tag me however so that I can review the generated driver portion though!

**Comment by kcxt** [comment]
Hey Marijn, I've also not dealt with the generator much before this... I don't think I can take responsibility for this whole feature right now.

I'll point it out to upstream at least.

**Comment by MarijnS95** [comment]
@calebccff Alright then, I'll see if I can open up a PR for this and see where we go from there. I haven't dealt much with the generator either, except being sent many many broken panel drivers from Konrad with a request to figure out why the panel isn't working, that all turned out to be generated :disappointed:

File: mipi.py [hunk 1]
```diff
@@ -218,6 +218,11 @@ def _generate_peripheral(t: Transaction, payload: bytes, options: Options) -> st
 	else:
 		raise ValueError(t)
 
+def _generate_dsc(t: Transaction, payload: bytes, options: Options) -> str:
+	if t != Transaction.DCS_COMPRESSION_MODE:
+		raise ValueError(t)
+	
+	return _generate_checked_call('mipi_dsi_compression_mode', ['dsi', 'true'], t.description)
```
Comment (MarijnS95) [inline mipi.py:225]: This is going to conflict with upstream https://github.com/msm8916-mainline/linux-mdss-dsi-panel-driver-generator/commit/7cb9fb2de0fbdff6342295aab65e55dd38f4a155, so I'll have to drop this when inevitably being forced to rebase for a successful upstream merge to take place :grin:

File: panel.py [hunk 2]
```diff
@@ -300,13 +300,13 @@ def __init__(self, name: str, fdt: Fdt2, node: int) -> None:
...
-			self.dsc_dsc_block_prediction = fdt.getprop_or_none(mode_node, 'qcom,mdss-dsc-block-prediction-enable')
+			self.dsc_dsc_block_prediction = fdt.getprop_or_none(mode_node, 'qcom,mdss-dsc-block-prediction-enable') != None
 			
 			dsc_version = fdt.getprop_or_none(mode_node, 'qcom,mdss-dsc-version')
-			self.dsc_version = dsc_version.as_int32() if dsc_version else None
+			self.dsc_version = dsc_version.as_int32() if dsc_version else 1
 
 			dsc_scr_version = fdt.getprop_or_none(mode_node, 'qcom,mdss-dsc-scr-version')
-			self.dsc_scr_version = dsc_scr_version.as_int32() if dsc_scr_version else None
+			self.dsc_scr_version = dsc_scr_version.as_int32() if dsc_scr_version else 1
```
Comment (MarijnS95) [inline panel.py:309]: Note to self: might want to change these to `as_uint32()` at some point, see upstream https://github.com/msm8916-mainline/linux-mdss-dsi-panel-driver-generator/commit/91b53842f9e4575467d7cb7d26b044c37dd2cc5a

**Review by MarijnS95 — APPROVED** [review]

**Comment by Saikatsaha1996** [comment]
I have dsc `cmd panel`.. so yesterday I foked your repository.. but unable to generate panel.. mey be still it is in early stage.. I unable to connect with you.. so I am sharing error here.. my error is..
```
Parsing: panel.dtb
Parsing: oplus21617samsung_ams662zs01_1080_2400_cmd_dvt (samsung ams662zs01 dvt dsc cmd mode panel)
Traceback (most recent call last):
  File "/root/linux-mdss-dsi-panel-driver-generator/./lmdpdg.py", line 77, in <module>
    panel = Panel.parse(fdt, offset)
  File "/root/linux-mdss-dsi-panel-driver-generator/panel.py", line 346, in parse
    return name and Panel(name.as_str(), fdt, node)
  File "/root/linux-mdss-dsi-panel-driver-generator/panel.py", line 295, in __init__
    dsi_ctrl = fdt.node_offset_by_phandle(dsi_ctrl.as_uint32())
  File "/usr/local/lib/python3.10/dist-packages/libfdt.py", line 529, in node_offset_by_phandle
    return check_err(fdt_node_offset_by_phandle(self._fdt, phandle), quiet)
  File "/usr/local/lib/python3.10/dist-packages/libfdt.py", line 129, in check_err
    raise FdtException(val)
libfdt.FdtException: pylibfdt error -1: FDT_ERR_NOTFOUND
panel.dtb does not contain any usable panel specifications
```
## My panel generation commd is 
`./lmdpdg.py -r vddio --backlight-gpio my_panel.dtb`

@z3ntu help me for find out your repository to me for my dsc panel.. I am attaching my panel here so I can understand about my panel..

https://github.com/Saikatsaha1996/realme-porsche/blob/main/dsi-panel-oplus21617samsung-ams662zs01-1080-2400-dvt-dsc-cmd.dts

I don't know C/C++ or python so I can't create or modify anything..

Thank you
Regards

**Comment by z3ntu** [comment]
Not really a good place to help but I'll try nevertheless

I don't think your dtb is the full one for your device. Best run this on a rooted device: `adb root && adb pull /sys/firmware/fdt your-device.dtb` , then `your-device.dtb` is the full dtb that's running on the device.

**Comment by Saikatsaha1996** [comment]
> Not really a good place to help but I'll try nevertheless
> 
> I don't think your dtb is the full one for your device. Best run this on a rooted device: `adb root && adb pull /sys/firmware/fdt your-device.dtb` , then `your-device.dtb` is the full dtb that's running on the device.

Yes it is not my full dtb.. it is only my panel dtsi file.. I just create build as dtb....
I have 60HZ, 90Hz, 120Hz timings.. so I removed 90Hz & 120 Hz timings.. and others I added to add my dts..

This is my downstream panel I create dtb with it..dtsi..https://github.com/Saikatsaha1996/android_kernel_oneplus_sm8350/blob/lineage-20/arch/arm64/boot/dts/vendor/oplus/porsche/dsi-panel-oplus21617-samsung-ams662zs01-1080-2400-dvt-dsc-cmd.dtsi

Why I am not creating dtb with my fdt.. because in my fdt has too many panels & timings so I used my own downstream panel dtsi ..

**Comment by MarijnS95** [comment]
This is the wrong place to discuss. Please use the upstream PR https://github.com/msm8916-mainline/linux-mdss-dsi-panel-driver-generator/pull/25 to discuss the effects of these changes as a whole.

Also, don't fork if you don't intend to contribute. Just clone. Your fork was way behind my branch yesterday and will inherently generate a faulty driver, if it compiles at all.

**Pull request merged**